runtime/src/test/java/org/corfudb/runtime/view/ClusterHealthTest.java

```diff
@@ -0,0 +1,117 @@
+
+    /**
+     * Test possible cluster statuses for layout servers
+     */
+    @Test
+    public void testLayoutServersHealth(){
+        Layout layout = layoutUtil.getLayout(servers);
+        ClusterStatus status = clusterHealth.getLayoutServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.STABLE);
+
+        layout.setUnresponsiveServers(Collections.singletonList(server3));
+        status = clusterHealth.getLayoutServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.DEGRADED);
+
+        layout.setUnresponsiveServers(Arrays.asList(server2, server3));
+        status = clusterHealth.getLayoutServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.UNAVAILABLE);
+    }
+
+    /**
+     * Test possible cluster statuses for sequencer servers
+     */
+    @Test
+    public void testSequencerServersHealth(){
+        Layout layout = layoutUtil.getLayout(servers);
+        layout.setUnresponsiveServers(Collections.singletonList(server3));
+
+        ClusterStatus status = clusterHealth.getSequencerServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.STABLE);
+
+        //Unresponsive sequencer
+        layout.setUnresponsiveServers(Collections.singletonList(server1));
+        status = clusterHealth.getSequencerServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.UNAVAILABLE);
+    }
+
+    /**
+     * Test possible cluster statuses for LogUnit servers
+     */
+    @Test
+    public void testLogUnitServersClusterHealth(){
+        Layout layout = layoutUtil.getLayout(servers);
+
+        ClusterStatus status = clusterHealth.getLogUnitServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.STABLE);
+
+        //invalid segment
+        layout.setUnresponsiveServers(Collections.singletonList(server3));
+        status = clusterHealth.getLogUnitServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.UNAVAILABLE);
+
+        //exclude unresponsive server
+        layout.getFirstSegment().getFirstStripe().getLogServers().remove(server3);
+        status = clusterHealth.getLogUnitServersClusterHealth(
+                layout, layout.getAllActiveServers()
+        );
+        assertThat(status).isEqualTo(ClusterStatus.STABLE);
+    }
+
+    /**
+     * Get cluster status from a layout with all responsive nodes.
+     * Expected result is STABLE status
+     */
+    @Test
+    public void testClusterHealth() {
```

Comment:
How different is this test from queryClusterStatus in ManagementViewTest ?

This test already exists and it actually is a bit more complete...

Reply:
That test is more an Integration Test, this one is a unit test. 
No need to start an entire cluster, no side effects from the environment, the unit test is much simpler. 
I think we need both - integration and unit tests even if some tests are similar.